lib/repositories/switch-repository.ts

```diff
@@ -0,0 +1,81 @@
+import { DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
+import Logger from 'bunyan';
+import { Entity, Table } from 'dynamodb-toolbox';
+
+import { DYNAMO_TABLE_KEY, DYNAMO_TABLE_NAME } from '../constants';
+import { SynthSwitchRequestBody } from '../handlers/synth-switch';
+import { BaseSwitchRepository } from './base';
+
+export const PARTITION_KEY = `${DYNAMO_TABLE_KEY.INPUT_TOKEN}#${DYNAMO_TABLE_KEY.INPUT_TOKEN_CHAIN_ID}${DYNAMO_TABLE_KEY.OUTPUT_TOKEN}#${DYNAMO_TABLE_KEY.OUTPUT_TOKEN_CHAIN_ID}#${DYNAMO_TABLE_KEY.TRADE_TYPE}`;
+
+export class SwitchRepository implements BaseSwitchRepository {
+  static log: Logger;
+
+  static create(documentClient: DynamoDBDocumentClient): BaseSwitchRepository {
+    this.log = Logger.createLogger({
+      name: 'DynamoSwitchRepository',
+      serializers: Logger.stdSerializers,
+    });
+
+    const switchTable = new Table({
+      name: DYNAMO_TABLE_NAME.SYNTH_SWITCH,
+      partitionKey: PARTITION_KEY,
+      sortKey: `${DYNAMO_TABLE_KEY.LOWER}`,
+      DocumentClient: documentClient,
+    });
+
+    const switchEntity = new Entity({
+      name: 'SynthSwitch',
+      attributes: {
+        [PARTITION_KEY]: { partitionKey: true },
+        lower: { sortKey: true },
+        enabled: { type: 'boolean' },
+      },
+      table: switchTable,
+      autoExecute: true,
+    } as const);
+
+    return new SwitchRepository(switchTable, switchEntity);
+  }
+
+  private constructor(
+    private readonly: Table<'SynthSwitch', 'inputToken', 'outputToken'>,
+    private readonly switchEntity: Entity
+  ) {}
+
+  public async syntheticQuoteForTradeEnabled(trade: SynthSwitchRequestBody): Promise<boolean> {
+    const { inputToken, inputTokenChainId, outputToken, outputTokenChainId, type, amount } = trade;
+
+    // get row for which lower bucket <= amount < upper bucket
+    const result = await this.switchEntity.query(
+      `${inputToken}#${inputTokenChainId}#${outputToken}#${outputTokenChainId}#${type}`,
+      {
+        limit: 1,
+        filters: [
+          { attr: 'lower', lte: amount },
+          { attr: 'upper', gt: amount },
```

Comment:
I imagine URA will call the `enabled` endpoint with native token amount (tokenIn amount if exact_input, etc), where it will be checked against the lower and upper bucket values, also in native amount, that are stored in our dynamo table 
@zhongeric let me know if that makes sense to you

Reply:
yeah this makes sense, URA should be calling with the lower and upper amounts in terms of the token, not USD. We should have the bucket logic in GPA for sure